### DISCUSSION TITLE
Photo of a sketch to perfect black and white line vector

### DISCUSSION THREAD

**Discussion by johnbrechbill** [ImageMagick/ImageMagick#5776] [Digital Image Processing]
Hello everyone, I am trying to take a photo of a sketch and turn that into what would look like a scanned sketch. I want the lines to be nearly perfect black and the paper to be nearly perfect white. Basically, I want to use the raspberry pi camera as a scanner. If there is anyway to do this in ImageMagick, please let me know. Thank you!

John

**Comment by fmw42** [comment]
Imagemagick does not have tools for scanning or importing images from your raspberry pi camera.  Once you have taken your image and have imported it into Imagemagick, it can make black into pure black and white into pure white via -threshold for example.

**Reply by johnbrechbill** [reply]
Got it, thank you!

**Comment by snibgo** [comment]
It may be possible to read the image from IM, using ffmpeg as a delegate. For example, on a laptop I have an entry in delegates.xml:
```
<delegate decode="webcm" command="&quot;ffmpeg.exe&quot; -f dshow -i video=&quot;HP Truevision HD&quot; -vframes %S -vcodec pam -an -f rawvideo -y &quot;%o&quot;"/>
```
ThenIM can read, say, frames 20 to 25:
```
magick webcm:[20-25] -resize 10% -append metimes6.jpg
```
But if you already have a binary that can read the camera, it may be easier to use that, either standalone or as an IM delegate.
